src/components/painter/painter.ts

```diff
@@ -253,26 +253,29 @@ export class ElementDrawArc extends ElementStatement {
      * Moves the sprite along arc with `radius` and `angle`.
      */
     onVisit(params: { [key: string]: TData }): void {
-        const radius = params['radius'] as number;
-        const angle = params['angle'] as number;
-
-        const direction = Math.sign(angle);
-        const theta = direction === 1 ? _state.heading + angle - 90 : _state.heading + angle + 90;
-
-        const [xCentre, yCentre] = [
-            _state.position.x + radius * Math.cos(degToRad(_state.heading + direction * 90)),
-            _state.position.y + radius * Math.sin(degToRad(_state.heading + direction * 90)),
-        ];
-
-        const [startAngle, stopAngle] =
-            direction === 1 ? [theta - angle, theta] : [theta, theta - angle];
-        sketch.drawArc(xCentre, yCentre, radius * 2, radius * 2, startAngle, stopAngle);
-
-        _state.heading += angle;
-        _state.position = {
-            x: xCentre + radius * Math.cos(degToRad(theta)),
-            y: yCentre + radius * Math.sin(degToRad(theta)),
-        };
+        if (_state.drawing) {
+            const radius = params['radius'] as number;
+            const angle = params['angle'] as number;
+
+            const direction = Math.sign(angle);
+            const theta =
+                direction === 1 ? _state.heading + angle - 90 : _state.heading + angle + 90;
+
+            const [xCentre, yCentre] = [
+                _state.position.x + radius * Math.cos(degToRad(_state.heading + direction * 90)),
+                _state.position.y + radius * Math.sin(degToRad(_state.heading + direction * 90)),
+            ];
+
+            const [startAngle, stopAngle] =
+                direction === 1 ? [theta - angle, theta] : [theta, theta - angle];
+            sketch.drawArc(xCentre, yCentre, radius * 2, radius * 2, startAngle, stopAngle);
```

Comment:
only put these lines inside the `if (_state.drawing)` check because the sprite still needs to move, just not draw anything